vfs/httpfs/union/union.go

```diff
@@ -0,0 +1,128 @@
+// Package union offers a simple http.FileSystem that can unify multiple filesystems at various mount points.
+package union
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/shurcooL/go/vfs/httpfs/vfsutil"
+)
+
+// New creates an union filesystem with the provided mapping of mount points to filesystems.
+//
+// Each mount point must be of form "/mydir". It must start with a '/', and contain a single directory name.
+func New(mapping map[string]http.FileSystem) http.FileSystem {
+	u := &unionFS{
```

Comment:
I'll think about this, but I'll keep it as is for now. This is already the style I use in many other places at this time, and when I was making the decision which way to go, I thought this way was better.

I've looked over sample source code in Go standard library now, and I'm seeing more often than not that structs are initialized as pointers and returned directly, rather than returning address of values.

Some examples:

- https://github.com/golang/go/blob/1045351cef21a64d954b4477af9f5105ea4287d3/src/net/http/transport.go#L568
- https://github.com/golang/go/blob/1045351cef21a64d954b4477af9f5105ea4287d3/src/net/http/request.go#L526

This may be rough and not very accurate. But since this decision affects more than just this code, I'd rather consider it outside of this PR. Thanks for mentioning it.